hazelcast/src/main/java/com/hazelcast/wan/WanReplicationPublisher.java

```diff
@@ -16,25 +16,67 @@
 
 package com.hazelcast.wan;
 
-import com.hazelcast.config.WANQueueFullBehavior;
+import com.hazelcast.config.AbstractWanPublisherConfig;
+import com.hazelcast.config.WanReplicationConfig;
+import com.hazelcast.internal.services.ServiceNamespace;
+import com.hazelcast.monitor.LocalWanPublisherStats;
+import com.hazelcast.spi.partition.PartitionReplicationEvent;
+
+import java.util.Collection;
+import java.util.Set;
 
 /**
- * This interface offers the implementation of different kinds of replication techniques like
- * TCP, UDP or maybe even an JMS based service
+ * This interface offers the implementation of different kinds of replication
+ * techniques like TCP, UDP or maybe even an JMS based service.
+ * Implementations of this interface represent a replication target,
+ * normally another Hazelcast cluster only reachable over a Wide Area
+ * Network (WAN).
+ * The publisher may implement {@link com.hazelcast.core.HazelcastInstanceAware}
+ * if it needs a reference to the instance on which it is being run.
+ *
+ * @param <T> WAN event container type (used for replication and migration inside the
+ *            cluster)
  */
-public interface WanReplicationPublisher {
+public interface WanReplicationPublisher<T> {
+    /**
+     * Initializes the publisher.
+     *
+     * @param wanReplicationConfig {@link WanReplicationConfig} instance
+     * @param publisherConfig      {@link AbstractWanPublisherConfig} instance
+     */
+    void init(WanReplicationConfig wanReplicationConfig, AbstractWanPublisherConfig publisherConfig);
+
+    /**
+     * Closes the publisher and its internal connections and shuts down other internal states.
+     * Signals the publisher to shut down and clean up its resources. The
+     * method does not necessarily block until the publisher has shut down.
+     */
+    void shutdown();
+
+    /**
+     * Resets the publisher (e.g. before split-brain merge).
+     * NOTE: used only in Hazelcast Enterprise.
+     */
+    default void reset() {
+    }
+
+    /**
+     * Prepares the publisher for event publication or performs pre-publication
+     * checks (e.g. enforcing invariants).
+     * Invoked before {@link #publishReplicationEvent(WanReplicationEvent)}
+     * and {@link #publishReplicationEventBackup(WanReplicationEvent)}.
+     */
+    void prepareForReplicationEventPublication();
```

Comment:
at first sight this method seems a similar method to `init`. Maybe another alternative is `doPrepublicationChecks`?

Reply:
Got it, will do.